Neos.ContentRepository.Core/Classes/Projection/ContentGraph/Node.php

```diff
@@ -75,9 +75,23 @@ public function __construct(
          * @return PropertyCollectionInterface Property values, indexed by their name
          */
         public PropertyCollectionInterface $properties,
+        /**
+         * The node's name. The name is guaranteed to be for tethered nodes.
+         * For the regular classification, it can be set optionally.
+         */
         public ?NodeName $nodeName,
         public Timestamps $timestamps,
     ) {
+        assert(
+            $this->classification !== NodeAggregateClassification::CLASSIFICATION_TETHERED
+            || $this->nodeName instanceof NodeName,
+            'The NodeName must be set if the Node is tethered.'
+        );
```

Comment:
One detail: [assert](https://www.php.net/manual/de/function.assert.php) is only evaluated if the corresponding PHP flag is set (and it's behavior is depending on those flags, too). So we should avoid it for productive code.
```suggestion
        if ($classification === NodeAggregateClassification::CLASSIFICATION_TETHERED && $nodeName === null) {
            throw new \InvalidArgumentException('The node name must be set if the node is tethered');
        }
```